docs/source/development/tools/xarray_to_cdf.rst

```diff
@@ -0,0 +1,239 @@
+      "FIELDNAM": "Energy bins",
+      "FILLVAL": np.float64(-1.0e31),
+      "FORMAT": "I12", # Display up to 12 numbers of an integer
+      "LABLAXIS": "Energy",
+      "UNITS": "eV",
+      "VALIDMIN": 1, #Fabricated value
+      "VALIDMAX": 3, #Fabricated value
+      "VAR_TYPE": "support_data",
+      "SCALETYP": "linear",
+      "VAR_NOTES": (
+         "We pretend there are 3 energy bins that data is collected for. "
+         "The 3 Energy bins are 0.5-1.5, 1.5-2.5, and 2.5-3.5 electron Volts. "
+         "These are used by the variable imap_jim_counts_(mode)_(description)"
+         "Ideally we would actually have additional variables that describe "
+         "the minimum and the maximum values in each of these bins, but for our "
+         "purposes of creating a minimal example, we'll leave extra varaibles out. "
+         "Longer notes about the variable can be placed into this attribute."
+      ),
+   }
+
+   # Describes a variable that holds the "counts" data
+   counts_attrs = {
+      "DEPEND_0": "Epoch",
+      "DEPEND_1": "Energy",
+      "DISPLAY_TYPE": "spectrogram",
+      "FILLVAL": INT_FILLVAL,
+      "FORMAT": "I12", # Display up to 12 numbers of an integer
+      "UNITS": "counts",
+      "VALIDMIN": 0, # Fabricated value
+      "VALIDMAX": 5000, # Fabricated value
+      "VAR_TYPE": "data",
+      "SCALETYP": "linear",
+      "VARIABLE_PURPOSE": "PRIMARY",
+      "CATDESC": "80 character description of the measurement goes here. For JIM, we mention it is counts per energy bin per time. ",
+      "FIELDNAM": "Counts per Energy Bin",
+      "LABLAXIS": "Ion Counts",
+      "VAR_NOTES": (
+         "The number of ions counted at each energy bin. "
+         "Sampled at 1 sample per second, at 1, 2, and 3 eV. "
+      ),
+
+   }
+
+   # Create 3 data arrays, one for time, one for the energies, and one for the real data
+   epoch_xr = xr.DataArray(
+                  name="Epoch",
+                  data=[np.datetime64("2025-01-01T00:00:01"), np.datetime64("2025-01-01T00:00:02"), np.datetime64("2025-01-01T00:00:03")],
+                  dims=("Epoch"),
+                  attrs=epoch_attrs,
+               )
+   energy_xr = xr.DataArray(
+                  name="Energy",
+                  data=[1,2,3], # Fabricated energy bins
+                  dims=("Energy_dim"),
+                  attrs=energy_bins_attrs,
+               )
+   counts_xr = xr.DataArray(
+               name="imap_jim_counts_(mode)_(descriptor)",
+               data=[[1,2,3], [4,5,6], [7,8,9]], # Fabricated data
+               dims=("Epoch", "Energy_dim"),
+               attrs=counts_attrs,
+         )
+   # Combine the data into an xarray.Dataset object
+   data = xr.Dataset(data_vars={
+                           "imap_jim_counts_mode_descriptor": counts_xr,
+                     },
+                     coords={
+                           "Epoch": epoch_xr,
+                           "Energy": energy_xr
```

Comment:
Does coordinates look up by 'name' and not the `dims`? The reason I ask is I noticed that the coordinate's name and `xr.DataArray`'s `dims` name did not match but it match with `xr.DataArray`'s `name`.

Reply:
In general, the coordinate's name and the dimension's name are separate things.   You could name them whatever you wanted for our purposes, as long as you are consistent within the xarray object.  The actual dimension names ("energy_dim" in this case) doesn't show up anywhere inside the CDF file, so it's purely for the xarray Dataset to link the two variables together.  

I admit it's a bit confusing how I set it up, because I just named one dimension "Epoch", but I added a "_dim" to  "Energy".  But my goal in doing this was to show that it doesn't ultimately matter what you name the dimensions.

Reply:
Thank you for clarifying this!